string name
uint16 samples
---
bool success